### PR TITLE
Ios build

### PR DESCRIPTION
**Pull request by CS6**
增加 ios 環境與設定

### CODE REVIEW

**Comment by bafu** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 16
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/numbersprotocol/capture-lite/pullRequest?prid=6208975&bid=20249226)

File: ios/App/App/AppDelegate.swift [hunk 1]
```diff
@@ -0,0 +1,74 @@
+import UIKit
+import Capacitor
+
+@UIApplicationMain
+class AppDelegate: UIResponder, UIApplicationDelegate {
+
+  var window: UIWindow?
+
+
+  func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
```
Comment (bafu) [inline ios/App/App/AppDelegate.swift:10]: Codacy found an issue: [Line should be 120 characters or less: currently 143 characters](https://app.codacy.com/gh/numbersprotocol/capture-lite/pullRequest?prid=6208975)

File: ios/App/App/AppDelegate.swift [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+  }
+
+  #if USE_PUSH
+
+  func application(_ application: UIApplication, didRegisterForRemoteNotificationsWithDeviceToken deviceToken: Data) {
+    NotificationCenter.default.post(name: Notification.Name(CAPNotifications.DidRegisterForRemoteNotificationsWithDeviceToken.name()), object: deviceToken)
+  }
+
+  func application(_ application: UIApplication, didFailToRegisterForRemoteNotificationsWithError error: Error) {
+    NotificationCenter.default.post(name: Notification.Name(CAPNotifications.DidFailToRegisterForRemoteNotificationsWithError.name()), object: error)
```
Comment (bafu) [inline ios/App/App/AppDelegate.swift:68]: Codacy found an issue: [Line should be 120 characters or less: currently 149 characters](https://app.codacy.com/gh/numbersprotocol/capture-lite/pullRequest?prid=6208975)

File: ios/App/App/AppDelegate.swift [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    // Use this method to pause ongoing tasks, disable timers, and invalidate graphics rendering callbacks. Games should use this method to pause the game.
+  }
+
+  func applicationDidEnterBackground(_ application: UIApplication) {
+    // Use this method to release shared resources, save user data, invalidate timers, and store enough application state information to restore your application to its current state in case it is terminated later.
+    // If your application supports background execution, this method is called instead of applicationWillTerminate: when the user quits.
+  }
+
+  func applicationWillEnterForeground(_ application: UIApplication) {
+    // Called as part of the transition from the background to the active state; here you can undo many of the changes made on entering the background.
```
Comment (bafu) [inline ios/App/App/AppDelegate.swift:26]: Codacy found an issue: [Line should be 120 characters or less: currently 151 characters](https://app.codacy.com/gh/numbersprotocol/capture-lite/pullRequest?prid=6208975)

File: ios/App/App/AppDelegate.swift [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+
+  func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
+    // Override point for customization after application launch.
+    return true
+  }
+
+  func applicationWillResignActive(_ application: UIApplication) {
+    // Sent when the application is about to move from active to inactive state. This can occur for certain types of temporary interruptions (such as an incoming phone call or SMS message) or when the user quits the application and it begins the transition to the background state.
+    // Use this method to pause ongoing tasks, disable timers, and invalidate graphics rendering callbacks. Games should use this method to pause the game.
```
Comment (bafu) [inline ios/App/App/AppDelegate.swift:17]: Codacy found an issue: [Line should be 120 characters or less: currently 155 characters](https://app.codacy.com/gh/numbersprotocol/capture-lite/pullRequest?prid=6208975)

File: ios/App/App/AppDelegate.swift [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+  }
+
+  func applicationDidEnterBackground(_ application: UIApplication) {
+    // Use this method to release shared resources, save user data, invalidate timers, and store enough application state information to restore your application to its current state in case it is terminated later.
+    // If your application supports background execution, this method is called instead of applicationWillTerminate: when the user quits.
```
Comment (bafu) [inline ios/App/App/AppDelegate.swift:22]: Codacy found an issue: [Line should be 120 characters or less: currently 137 characters](https://app.codacy.com/gh/numbersprotocol/capture-lite/pullRequest?prid=6208975)

File: ios/App/App/AppDelegate.swift [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+  func applicationWillTerminate(_ application: UIApplication) {
+    // Called when the application is about to terminate. Save data if appropriate. See also applicationDidEnterBackground:.
+  }
+
+  func application(_ app: UIApplication, open url: URL, options: [UIApplication.OpenURLOptionsKey : Any] = [:]) -> Bool {
+    // Called when the app was launched with a url. Feel free to add additional processing here,
+    // but if you want the App API to support tracking app url opens, make sure to keep this call
+    return CAPBridge.handleOpenUrl(url, options)
+  }
+
```
Comment (bafu) [inline ios/App/App/AppDelegate.swift:42]: Codacy found an issue: [Lines should not have trailing whitespace.](https://app.codacy.com/gh/numbersprotocol/capture-lite/pullRequest?prid=6208975)

File: ios/App/App/AppDelegate.swift [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+  func application(_ application: UIApplication, didFailToRegisterForRemoteNotificationsWithError error: Error) {
+    NotificationCenter.default.post(name: Notification.Name(CAPNotifications.DidFailToRegisterForRemoteNotificationsWithError.name()), object: error)
+  }
+
+#endif
+
+}
+
```
Comment (bafu) [inline ios/App/App/AppDelegate.swift:74]: Codacy found an issue: [Files should have a single trailing newline.](https://app.codacy.com/gh/numbersprotocol/capture-lite/pullRequest?prid=6208975)

File: ios/App/App/AppDelegate.swift [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+  func applicationDidBecomeActive(_ application: UIApplication) {
+    // Restart any tasks that were paused (or not yet started) while the application was inactive. If the application was previously in the background, optionally refresh the user interface.
+  }
+
+  func applicationWillTerminate(_ application: UIApplication) {
+    // Called when the application is about to terminate. Save data if appropriate. See also applicationDidEnterBackground:.
+  }
+
+  func application(_ app: UIApplication, open url: URL, options: [UIApplication.OpenURLOptionsKey : Any] = [:]) -> Bool {
```
Comment (bafu) [inline ios/App/App/AppDelegate.swift:37]: Codacy found an issue: [Colons should be next to the identifier when specifying a type and next to the key in dictionary literals.](https://app.codacy.com/gh/numbersprotocol/capture-lite/pullRequest?prid=6208975)

File: ios/App/App/AppDelegate.swift [hunk 9]
```diff
@@ -0,0 +1,74 @@
+import UIKit
+import Capacitor
+
+@UIApplicationMain
+class AppDelegate: UIResponder, UIApplicationDelegate {
+
+  var window: UIWindow?
+
+
```
Comment (bafu) [inline ios/App/App/AppDelegate.swift:9]: Codacy found an issue: [Limit vertical whitespace to a single empty line. Currently 2.](https://app.codacy.com/gh/numbersprotocol/capture-lite/pullRequest?prid=6208975)

File: ios/App/App/AppDelegate.swift [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+  func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
+    // Override point for customization after application launch.
+    return true
+  }
+
+  func applicationWillResignActive(_ application: UIApplication) {
+    // Sent when the application is about to move from active to inactive state. This can occur for certain types of temporary interruptions (such as an incoming phone call or SMS message) or when the user quits the application and it begins the transition to the background state.
```
Comment (bafu) [inline ios/App/App/AppDelegate.swift:16]: Codacy found an issue: [Line should be 120 characters or less: currently 281 characters](https://app.codacy.com/gh/numbersprotocol/capture-lite/pullRequest?prid=6208975)

File: ios/App/App/AppDelegate.swift [hunk 11]
```diff
@@ -0,0 +1,74 @@
...
+
+    if statusBarRect.contains(touchPoint) {
+      NotificationCenter.default.post(CAPBridge.statusBarTappedNotification)
+    }
+  }
+
+  #if USE_PUSH
+
+  func application(_ application: UIApplication, didRegisterForRemoteNotificationsWithDeviceToken deviceToken: Data) {
+    NotificationCenter.default.post(name: Notification.Name(CAPNotifications.DidRegisterForRemoteNotificationsWithDeviceToken.name()), object: deviceToken)
```
Comment (bafu) [inline ios/App/App/AppDelegate.swift:64]: Codacy found an issue: [Line should be 120 characters or less: currently 155 characters](https://app.codacy.com/gh/numbersprotocol/capture-lite/pullRequest?prid=6208975)

File: ios/App/App/AppDelegate.swift [hunk 12]
```diff
@@ -0,0 +1,74 @@
...
+    return true
+  }
+
+  func applicationWillResignActive(_ application: UIApplication) {
+    // Sent when the application is about to move from active to inactive state. This can occur for certain types of temporary interruptions (such as an incoming phone call or SMS message) or when the user quits the application and it begins the transition to the background state.
+    // Use this method to pause ongoing tasks, disable timers, and invalidate graphics rendering callbacks. Games should use this method to pause the game.
+  }
+
+  func applicationDidEnterBackground(_ application: UIApplication) {
+    // Use this method to release shared resources, save user data, invalidate timers, and store enough application state information to restore your application to its current state in case it is terminated later.
```
Comment (bafu) [inline ios/App/App/AppDelegate.swift:21]: Codacy found an issue: [Line should be 120 characters or less: currently 214 characters](https://app.codacy.com/gh/numbersprotocol/capture-lite/pullRequest?prid=6208975)

File: ios/App/App/AppDelegate.swift [hunk 13]
```diff
@@ -0,0 +1,74 @@
...
+    // If your application supports background execution, this method is called instead of applicationWillTerminate: when the user quits.
+  }
+
+  func applicationWillEnterForeground(_ application: UIApplication) {
+    // Called as part of the transition from the background to the active state; here you can undo many of the changes made on entering the background.
+  }
+
+  func applicationDidBecomeActive(_ application: UIApplication) {
+    // Restart any tasks that were paused (or not yet started) while the application was inactive. If the application was previously in the background, optionally refresh the user interface.
```
Comment (bafu) [inline ios/App/App/AppDelegate.swift:30]: Codacy found an issue: [Line should be 120 characters or less: currently 190 characters](https://app.codacy.com/gh/numbersprotocol/capture-lite/pullRequest?prid=6208975)

File: ios/App/App/AppDelegate.swift [hunk 14]
```diff
@@ -0,0 +1,74 @@
...
+  }
+
+  func applicationWillTerminate(_ application: UIApplication) {
+    // Called when the application is about to terminate. Save data if appropriate. See also applicationDidEnterBackground:.
```
Comment (bafu) [inline ios/App/App/AppDelegate.swift:34]: Codacy found an issue: [Line should be 120 characters or less: currently 124 characters](https://app.codacy.com/gh/numbersprotocol/capture-lite/pullRequest?prid=6208975)

File: ios/App/App/AppDelegate.swift (hunk 8 above)
Comment (bafu) [inline ios/App/App/AppDelegate.swift:37]: Codacy found an issue: [Line should be 120 characters or less: currently 121 characters](https://app.codacy.com/gh/numbersprotocol/capture-lite/pullRequest?prid=6208975)

File: ios/App/App/AppDelegate.swift [hunk 15]
```diff
@@ -0,0 +1,74 @@
...
+  }
+
+  func application(_ app: UIApplication, open url: URL, options: [UIApplication.OpenURLOptionsKey : Any] = [:]) -> Bool {
+    // Called when the app was launched with a url. Feel free to add additional processing here,
+    // but if you want the App API to support tracking app url opens, make sure to keep this call
+    return CAPBridge.handleOpenUrl(url, options)
+  }
+  
+  func application(_ application: UIApplication, continue userActivity: NSUserActivity, restorationHandler: @escaping ([UIUserActivityRestoring]?) -> Void) -> Bool {
```
Comment (bafu) [inline ios/App/App/AppDelegate.swift:43]: Codacy found an issue: [Line should be 120 characters or less: currently 165 characters](https://app.codacy.com/gh/numbersprotocol/capture-lite/pullRequest?prid=6208975)

**Comment by CS6** [comment]
@bafu   PR Quality Review 對於 Ionic 自動建立的 ios swift 的 issue 需要處理嗎？

**Comment by seanwu1105** [comment]
> @bafu PR Quality Review 對於 Ionic 自動建立的 ios swift 的 issue 需要處理嗎？

@CS6 這個 Quality Review 是我建立的，不過我沒有針對 Swift 做 linting 的設定，你可以簡單參考或是直接進 Codacy 修改 linting 規定即可。

File: package.json [hunk 16]
```diff
@@ -21,15 +21,20 @@
     "@angular/platform-browser": "~10.0.6",
     "@angular/platform-browser-dynamic": "~10.0.6",
     "@angular/router": "~10.0.6",
+
     "@capacitor/android": "^2.4.1",
     "@capacitor/core": "2.4.1",
+    "@capacitor/ios": "^2.4.0",
```
Comment (shc261392) [inline package.json:27]: @capacitor/ios should be bumped to 2.4.1 as well

**Review by shc261392 — CHANGES_REQUESTED** [review]

**Review by shc261392 — APPROVED** [review]

**Pull request merged**